web-common/src/features/custom-dashboards/Component.svelte

```diff
@@ -26,16 +29,25 @@
   export let interacting = false;
   export let width: number;
   export let height: number;
-  export let i: number;
   export let localZIndex = 0;
   export let chartView = false;
+  export let componentName: string;
+  export let instanceId: string;
+
+  $: resourceQuery = useResource(
+    instanceId,
```

Comment:
Whats the idea of getting the instance id in the params? We have the global to avoid prop drilling.

Reply:
Nothing hard and fast here. Just an instinct to get implementation details out of what are essentially display components. The runtime global is likely to change structure in the future. It also just means that only one component is subscribing to the store as opposed to however many dashboard elements there are. Not critical, though.